shell/src/main/java/alluxio/cli/fsadmin/report/SummaryCommand.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.fsadmin.report;
+
+import alluxio.client.block.RetryHandlingBlockMasterClient;
+import alluxio.client.MetaMasterClient;
+import alluxio.util.CommonUtils;
+import alluxio.util.FormatUtils;
+import alluxio.wire.BlockMasterInfo;
+import alluxio.wire.BlockMasterInfo.BlockMasterInfoField;
+import alluxio.wire.MasterInfo;
+import alluxio.wire.MasterInfo.MasterInfoField;
+
+import com.google.common.base.Strings;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Prints Alluxio cluster summarized information.
+ */
+public class SummaryCommand {
+  private static final int INDENT_SIZE = 4;
+
+  private MetaMasterClient mMetaMasterClient;
+  private RetryHandlingBlockMasterClient mBlockMasterClient;
+  private int mIndentationLevel = 1;
+
+  /**
+   /**
+   * Creates a new instance of {@link SummaryCommand}.
+   *
+   * @param metaMasterClient client to connect to meta master
+   * @param blockMasterClient client to connect to block master
+   */
+  public SummaryCommand(MetaMasterClient metaMasterClient,
```

Comment:
can we add a unit test for this class? You can mock the master interactions, and pass a `PrintWriter` to the constructor to see what the class prints

Reply:
The SummaryCommandIntegrationTest.java is the unit test for SummaryCommand.
In SummaryCommand, I do not use PrintWriter to print the user-facing information. Would it be better to use PrintWriter to save the print out information and read the content from the file and print out at the end?

Reply:
Unit tests and integration tests are two different types of tests. Unit tests focus on a small part of the code and mock the interactions with the rest of the system. Integration tests run the whole system and check that everything works together. Unit tests are fast and catch problems isolated to a particular component. Integration tests are slower and can catch problems with the integration between different components of the system.

Reply:
I use two ways to test the SummaryCommand.
One is to use mFsAdminShell.run("report", "summary");
Another call SummaryCommand with dependencies directly, following is the code:
```java
  @Test
  public void callSummaryWithDependencies() throws IOException {
    try (MetaMasterClient mMetaMasterClient
        = new RetryHandlingMetaMasterClient(MasterClientConfig.defaults());
        RetryHandlingBlockMasterClient mBlockMasterClient = new RetryHandlingBlockMasterClient(
        MasterClientConfig.defaults())) {
      SummaryCommand summaryCommand = new SummaryCommand(mMetaMasterClient, mBlockMasterClient);
      summaryCommand.run();
      String output = mOutput.toString();
      checkSummaryResults(output);
    }
  }
```